core/modules/parser/SelectParser.cc

```diff
@@ -237,6 +237,50 @@ class Antlr2Parser : public AntlrParser {
 };
 
 
+class Antlr4ErrorStrategy : public antlr4::DefaultErrorStrategy {
+public:
+    Antlr4ErrorStrategy(std::string const& statement) : _statement(statement) {}
+
+private:
+    void recover(antlr4::Parser *recognizer, std::exception_ptr e) override {
+        LOGS(_log, LOG_LVL_ERROR, __FUNCTION__ <<
+                " antlr4 could not make a parse tree out of the input statement:" << _statement);
+        throw ParseException(std::string("Failed to instantiate query: \"") + _statement +
+                std::string("\""));
+    }
+
+    antlr4::Token* recoverInline(antlr4::Parser *recognizer) override {
+        LOGS(_log, LOG_LVL_ERROR, __FUNCTION__ <<
+                " antlr4 could not make a parse tree out of the input statement:" << _statement);
+        throw ParseException(std::string("Failed to instantiate query: \"") + _statement +
+                std::string("\""));
+    }
+
+    void sync(antlr4::Parser *recognizer) override {
+        // we want this function to be a no-op so we override it.
+    }
+
+    std::string const& _statement;
+};
+
+
+class NonRecoveringQSMySqlLexer : public QSMySqlLexer {
+public:
+    NonRecoveringQSMySqlLexer(antlr4::CharStream *input, std::string const & statement)
+        : QSMySqlLexer(input), _statement(statement) {}
+
+private:
+    virtual void recover(const antlr4::LexerNoViableAltException &e) {
```

Comment:
Names of the private methods are supposed to begin with underscore ...unless you're overriding a virtual method which isn't under your control (comes from ANTLR4)